### DISCUSSION TITLE
[Bug] Explicit layout with record struct

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/roslyn#66372] [General]
For example
```F#
[<Struct; StructLayout(LayoutKind.Explicit)>]
 type Dimension2D = 
            {
                [<FieldOffsetAttribute(0)>]
                Memory : Vector2
                [<FieldOffsetAttribute(0)>]
                Width : float32
                [<FieldOffsetAttribute(4)>]
                Height : float32
            }
            with
            static member Zero = { Memory = Vector2.Zero; Width = 0f; Height = 0f; }
```
The point of explicit layout is you can change the memory by changing width / height, or the opposite way by using System.Numerics.Vector2 methods to change the memory

However I noticed that, while it's already hard to create a record with only memory is set
```F#
{Dimension2D.Memory = Vector2.Zero} //Alone doesn't work
```
If I were to do :
```F#
{ Dimension2D.Zero with Memory = Vector2.Multiply(aVector, value) } //Bug : the value remains zero after this no matter what you do
```
Where ```AnotherMemory``` is not zero and so is value, the result is zero for some reason

But if I were to do
```F#
let v = Vector2.Multiply(aVector, value)
{ Dimension2D.Zero with Memory = v; Width = aVector.X; Height = aVector.Y } //Works
```

It gives me a feeling that either I can't have record struct with explicit layout in F# like I can in C#, or there is a bug with it

**Comment by CyrusNajmabadi** [comment]
Are you asking a question about f#?

**Reply by Xyncgas** [reply]
My question is, how is this consistent with F# and C# both being Microsoft's .NET's language but the ```with``` keyword works differently, if there is clearly designs that are intentional to be inconsistent then the people writing compiler for these languages might have an answer, and if it's unintentional, I would be looking forward to hearing if you consider this as a bug and fix it and if so is C# way the right way to do it or should the ```with``` keyword works like how it's working in F# for the effect it's having on explicit layout

**Reply by CyrusNajmabadi** [reply]
I don't see any c# in your post. So I'm not sure what you're comparing to.
